### PR TITLE
[SPARK-36851][SQL] Incorrect parsing of negative ANSI typed interval literals

### PR DESCRIPTION
**Pull request by Peng-Lei**
### What changes were proposed in this pull request?
Handle incorrect parsing of negative ANSI typed interval literals
[SPARK-36851](https://issues.apache.org/jira/browse/SPARK-36851)

### Why are the changes needed?
Incorrect result:
```
spark-sql> select interval -'1' year;
1-0
```

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Add ut testcase

### CODE REVIEW

**Comment by Peng-Lei** [comment]
@MaxGekk Could you take a look ? Is this fix okay ? Thank you.

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala [hunk 1]
```diff
@@ -4223,6 +4223,17 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     checkAnswer(sql("""SELECT from_json(r'{"a": "\\"}', 'a string')"""), Row(Row("\\")))
     checkAnswer(sql("""SELECT from_json(R'{"a": "\\"}', 'a string')"""), Row(Row("\\")))
   }
+
+  test("SPARK-36851: Incorrect parsing of negative ANSI typed interval literals") {
+    val df1 = sql("select interval -'1' year")
```
Comment (gengliangwang) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala:4228]: Let's move the test cases into `interval.sql`.
Also, we should add tests for negative day time intervals as well.

**Comment by gengliangwang** [comment]
I will cut RC5 after merging this one.

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala (hunk 1 above)
Comment (Peng-Lei) [reply sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala:4228]: OK

Comment (Peng-Lei) [reply sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala]: done

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master/3.2

**Comment by cloud-fan** [comment]
good catch!

**Review by MaxGekk — COMMENTED** [review]
I have only the concern about the place you fixed. You changed the code related to multi units intervals parsing but I assume ANSI intervals (unit to unit) in my example in the JIRA ticket.

**Pull request closed without merging**